.github/workflows/scripts/slack_notifications/gem_notifier.rb

```diff
@@ -0,0 +1,83 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'time'
+require 'httparty'
+require_relative 'slack_notifier'
+
+class GemNotifier < SlackNotifier
+  def self.check_for_updates(watched_gems)
+    return if gem_list_empty(watched_gems)
+    watched_gems.each do |gem_name|
+      verify_gem(gem_name) ? gem_info = verify_gem(gem_name) : return
+      versions = gem_versions(gem_info)
+      notifier.send_slack_message(gem_message(gem_name, versions)) if gem_updated?(versions)
+    end
+  end
+
+  def self.gem_list_empty(watched_gems)
+    if watched_gems.empty?
+      abort("Nothing to see here! The 'watched_gems' array cannot be empty")
+    end
```

Comment:
Looking at the name `gem_list_empty`, it's really tempting to say "You should rename this to `gem_list_empty?` so that we know that we're performing an emptiness check". But then you might respond with "But it won't ever return `false`, it's more of a verification process".

At which point I'd then propose we use a different naming scheme entirely. "Can we rename it to `verify_gem_list` or `verify_gem_list!` then? As-is, it sounds like you want the program to take hold of something full and go and empty it."

I'd one-liner the contents of the method too:

```ruby
abort("Nothing to see here! The 'watched_gems' array cannot be empty") if watched_gems.empty?
```

Reply:
I see what you're saying about the naming. I would think the same thing just reading it now. [Updated](https://github.com/newrelic/newrelic-ruby-agent/pull/1432/commits/6b044062a798dca30a0c6649b73684c31246f934).